homeassistant/components/mqtt/__init__.py

```diff
@@ -1273,8 +1270,15 @@ def device_info(self):
     """Delete device."""
     device_id = msg["device_id"]
     dev_registry = await get_dev_reg(hass)
-    dev_registry.async_remove_device(device_id)
-    connection.send_message(websocket_api.result_message(msg["id"]))
+
+    device = dev_registry.async_get(device_id)
+    for config_entry in device.config_entries:
+        config_entry = hass.config_entries.async_get_entry(config_entry)
+        # Only delete the device if it belongs to an MQTT device entry
+        if config_entry.domain == DOMAIN:
+            dev_registry.async_remove_device(device_id)
+            connection.send_message(websocket_api.result_message(msg["id"]))
+            break
```

Comment:
You're not sending an error if you can't find the device .

Reply:
Right, but isn't it enough to not send success?

Reply:
No. This will cause the test to never finish if you're awaiting a response.